### PR TITLE
Handle empty meta/main.yml in RoleNames rule

### LINKED ISSUE #1021: Empty meta causes "'NoneType' object is not subscriptable"
##### Summary

Empty meta/main.yml causes the following error:
```
Traceback (most recent call last):
  File ".../python3.8/site-packages/ansiblelint/rules/RoleNames.py", line 62, in match
    role_name = parse_yaml_from_file(str(meta))['galaxy_info']['role_name']
TypeError: 'NoneType' object is not subscriptable
```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible 2.9.6
  config file = None
  configured module search path = ['.../.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = .../ansible-lint/venv/lib/python3.8/site-packages/ansible
  executable location = .../ansible-lint/venv/bin/ansible
  python version = 3.8.2 (default, Jul 16 2020, 14:00:26) [GCC 9.3.0]

ansible-lint --version
ansible-lint 4.3.3
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Ubuntu Linux 20.04

##### STEPS TO REPRODUCE

Lint a playbook with an empty _meta/main.yml_.

##### Desired Behaviour

Playbook should be linted without crash.

##### Actual Behaviour

ansible-lint crashes:

```
Traceback (most recent call last):
  File ".../python3.8/site-packages/ansiblelint/rules/RoleNames.py", line 62, in match
    role_name = parse_yaml_from_file(str(meta))['galaxy_info']['role_name']
TypeError: 'NoneType' object is not subscriptable
```

PR with test and fix will follow in a few minutes.

### PR DESCRIPTION
**Pull request by phihos**
The RoleNames rule tries to access the meta information of a role if meta/main.yml exists.
If its contents are empty `parse_yaml_from_file` returns `None`. That circumstance was not checked causing `'NoneType' object is not subscriptable`.

I added a simple check before the rule tries to access the role's meta information.

Fixes: #1021

### CODE REVIEW

File: test/TestRoleNames.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+}
+
+ROLE_WITH_EMPTY_META = {**ROLE_MINIMAL, **ROLE_META_EMPTY}
+
+PLAY_INCLUDE_ROLE = """
+- hosts: all
+  roles:
+    - {}
+""".format(ROLE_NAME_VALID)
```
Comment (webknjaz) [inline test/TestRoleNames.py:34]:
```suggestion
PLAY_INCLUDE_ROLE = f"""
- hosts: all
  roles:
    - {ROLE_NAME_VALID}
"""
```

File: test/TestRoleNames.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                     id='ROLE_EMPTY_META',
+                     ),
+), indirect=('playbook_path',))
+def test_role_name(playbook_path, messages):
+    """Lint a playbook and compore the expected messages with the actual messages."""
+    runner = Runner(RULES_COLLECTION, playbook_path, [], [], [])
+    results = runner.run()
+    for message in messages:
+        assert message in str(results)
+    assert len(results) == len(messages)
```
Comment (webknjaz) [inline test/TestRoleNames.py:74]: I'd put this before the loop as an early check

File: test/TestRoleNames.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                     [],
+                     id='ROLE_EMPTY_META',
+                     ),
+), indirect=('playbook_path',))
+def test_role_name(playbook_path, messages):
+    """Lint a playbook and compore the expected messages with the actual messages."""
+    runner = Runner(RULES_COLLECTION, playbook_path, [], [], [])
+    results = runner.run()
+    for message in messages:
+        assert message in str(results)
```
Comment (webknjaz) [inline test/TestRoleNames.py:73]: instead of applying `str()` on each iteration, do it before looping

File: test/TestRoleNames.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
+"""Test the RoleNames rule."""
+
+import pytest
+
+from ansiblelint.rules import RulesCollection
+from ansiblelint.rules.RoleNames import RoleNames
+from ansiblelint.runner import Runner
+
+RULES_COLLECTION = RulesCollection()
+RULES_COLLECTION.register(RoleNames())
```
Comment (webknjaz) [inline test/TestRoleNames.py:10]: Why is this necessary?

**Comment by webknjaz** [comment]
@phihos the PR title and description should be verbose about what this does.

File: test/TestRoleNames.py (hunk 4 above)
Comment (phihos) [reply test/TestRoleNames.py:10]: I just want to test the RoleNames rule. Should I do that differently?

**Comment by webknjaz** [comment]
Also, your first commit created on your computer does not have committer email that is verified in your GitHub profile.
It is likely that Git on your computer is misconfigured. You may want to use `git config --global user.email YOUR@GITHUB.EMAIL` or maybe add the email to your GitHub account.

File: test/TestRoleNames.py (hunk 4 above)
Comment (webknjaz) [reply test/TestRoleNames.py:10]: Where did you pick up this snippet? I don't recall any other tests needing register.

Comment (webknjaz) [reply test/TestRoleNames.py:10]: Ah... I now understand why it looks weird. Other places use fixtures for test deps: https://github.com/ansible/ansible-lint/blob/0dbdbbd8196326e1c0192667ba3674f3f9f88c46/test/TestRulesCollection.py#L32.

File: test/TestRoleNames.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+@pytest.mark.parametrize(('playbook_path', 'messages'), (
+        pytest.param((PLAY_INCLUDE_ROLE, ROLE_NAME_VALID, ROLE_WITH_EMPTY_META),
+                     [],
+                     id='ROLE_EMPTY_META',
+                     ),
+), indirect=('playbook_path',))
+def test_role_name(playbook_path, messages):
+    """Lint a playbook and compore the expected messages with the actual messages."""
+    runner = Runner(RULES_COLLECTION, playbook_path, [], [], [])
+    results = str(runner.run())
```
Comment (webknjaz) [inline test/TestRoleNames.py:71]: you probably need another var because the next line now checks the length of a string, not of a list as it was originally

File: test/TestRoleNames.py (hunk 4 above)
Comment (phihos) [reply test/TestRoleNames.py]: I refactored these lines to a fixture. Can you verify that it is the way you hinted at?

**Comment by phihos** [comment]
@webknjaz I think I addressed the issues you raised. Could you review again?

File: test/TestRoleNames.py (hunk 4 above)
Comment (phihos) [reply test/TestRoleNames.py]: Btw I got that register stuff from the setUp functions of the unittest test cases. Like `TestAlwaysRun`.

**Review by ssbarnea — APPROVED** [review]

**Comment by phihos** [comment]
The indentation of the test task was off so  I fixed it by externalizing the task definition to an of variable.

File: test/TestRoleNames.py (hunk 4 above)
Comment (webknjaz) [reply test/TestRoleNames.py]: Yeah, I'd rather get rid of unittest because those classes are incompatible  with pytest fixtures: https://github.com/ansible/ansible-lint/issues/725

**Review by webknjaz — APPROVED** [review]
